### PR TITLE
[Valkey] Replace Redis with Valkey

### PR DESCRIPTION
**Pull request by julianocosta89**
# Changes

This PR replaces Redis with Valkey.

Redis has recently changed its licensing and Valkey was created by the community to continue the project as an Open Source driven project.
[Valkey/about](https://valkey.io/about/):
> The transition garnered substantial backing from leading tech corporations including AWS, Google, Oracle, Ericsson, and Snap, with the Linux Foundation stepping in to provide a stable platform for this new direction.

[Linux Foundation Launches Open Source Valkey Community](https://www.linuxfoundation.org/press/linux-foundation-launches-open-source-valkey-community).

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* [x] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

File: src/otelcollector/otelcol-config.yml [hunk 1]
```diff
@@ -14,7 +14,8 @@ receivers:
     targets:
       - endpoint: http://frontendproxy:${env:ENVOY_PORT}
   redis:
```
Comment (rogercoll) [inline src/otelcollector/otelcol-config.yml:16]: I am wondering if the collector should rename this receiver to `datastore` or something more generic that envelopes valkey and redis

**Review by rogercoll — APPROVED** [review]
LGTM! Neat migration :clap:

File: src/otelcollector/otelcol-config.yml (hunk 1 above)
Comment (julianocosta89) [reply src/otelcollector/otelcol-config.yml:16]: yeah, I'd +1 that.
There are also the .NET dependencies and instrumentation that are still called Redis something.

**Review by puckpuck — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

**Comment by rogercoll** [comment]
Just opened this PR to update the helm charts: https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1232 

(nightly images are failing with helm deployment)

**Comment by sorenlouv** [comment]
Hey!

The cartservice is now seemingly unable to start and is throwing this error

> REDIS_ADDR environment variable is required.

**Pull request merged**